core/src/main/scala/io/delta/sql/parser/DeltaSqlParser.scala

```diff
@@ -346,7 +346,28 @@ class DeltaSqlAstBuilder extends DeltaSqlBaseBaseVisitor[AnyRef] {
     OptimizeTableCommand(
       Option(ctx.path).map(string),
       Option(ctx.table).map(visitTableIdentifier),
-      Option(ctx.partitionPredicate).map(extractRawText(_)).toSeq, Map.empty)(interleaveBy)
+      Option(ctx.partitionPredicate).map(extractRawText(_)).toSeq,
+      Map.empty)(interleaveBy)
+  }
+
+  /**
+   * Creates a [[ReorgTable]] logical plan.
+   * Examples:
+   * {{{
+   *   -- Physically delete dropped columns of target table
+   *   REORG TABLE (delta.`/path/to/table` | delta_table_name)
+   *    [WHERE partition_predicate] APPLY (PURGE)
+   * }}}
+   */
+  override def visitReorgTable(ctx: ReorgTableContext): AnyRef = withOrigin(ctx) {
+    if (ctx.table == null) {
+      throw new ParseException("REORG command requires a file path or table name.", ctx)
+    }
+
+    val targetIdentifier = visitMultipartIdentifier(ctx.table)
```

Comment:
why not use the `visitTableIdentifier`?

Reply:
It's mainly because `UnresolvedTable` accepts a multipart table name rather than a `TableIdentifier`. I figured out it's possible to assemble the multipart name  manually:
```
    val targetIdentifier = visitTableIdentifier(ctx.table)
    val tableNameParts = targetIdentifier.database.toSeq :+ targetIdentifier.table
```

Reply:
We use `UnresolvedTable` here so it can throw a nice analysis error when the target table is a temp view.